### PR TITLE
Improved Modal component

### PR DESCRIPTION
**Pull request by benja**
I've removed the Reach dependency and updated a few things to align the React version with [Vue](http://fabric-vue.surge.sh/modal).

### CODE REVIEW

**Comment by pearofducks** [comment]
I appreciate the c/p of wu-tang-ipsum :D

I don't see a scroll-locking solution implemented. Is there something about this implementation that makes that not necessary?

(scroll-locking is usually needed on iOS because the spring in overflow scrolling causes lots of issues - especially if the background (non-modal) content is also scrollable)

**Comment by benja** [comment]
> I don't see a scroll-locking solution implemented. Is there something about this implementation that makes that not necessary?

I thought a simple `overflow: hidden` on the `<body>` tag would do. Did not know that it's not supported by older iOS versions. I've gone ahead and added proper scroll locking now.

File: packages/modal/src/component.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+}: ModalProps) => {
+    const contentRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (!contentRef.current) return;
+        console.log(contentRef);
+        if (props.open) {
+            setup(contentRef.current);
+        } else {
+            teardown(contentRef.current);
```
Comment (pearofducks) [inline packages/modal/src/component.tsx:24]: Teardown doesn't take any args (I really should finish this module up and add types) 🙈

**Comment by pearofducks** [comment]
> Did not know that it's not supported by older iOS versions. I've gone ahead and added proper scroll locking now.

Even the current iOS version derps pretty hard on modals without scroll-locking, it's very 😢

File: packages/modal/src/component.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+export const Modal = ({
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+    ...props
+}: ModalProps) => {
+    const contentRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (!contentRef.current) return;
+        console.log(contentRef);
```
Comment (pearofducks) [inline packages/modal/src/component.tsx:20]: stray `console.log`?

File: packages/modal/src/component.tsx (hunk 1 above)
Comment (benja) [reply packages/modal/src/component.tsx]: Types would definitely help, haha. I first tried another scroll-lock package where you had to implicitly specify the element, then I switched to yours and forgot to remove the param. 😄 I've made a few changes after your feedback and it all seems fine now.

Comment (benja) [reply packages/modal/src/component.tsx]: Actually, it does not seem like the `teardown` is replacing scroll properly. I'll have a quick look at what's going on.

Comment (benja) [reply packages/modal/src/component.tsx]: Fixed.

**Review by pearofducks — APPROVED** [review]
lgtm!

**Pull request merged**